### PR TITLE
Merge 1.68 into cesium.com

### PR DESCRIPTION
**Pull request by OmarShehata**
This merges the 1.68 tag into cesium.com.

I had a merge conflict in `gulpfile.cjs` that I'd like a second set of eyes to confirm this looks right. As far as I can tell the way we upload files changed slightly since last month. I resolved the conflict by taking in the new changes from 1.68.

See:
https://github.com/CesiumGS/cesium/pull/8715/files#diff-c747526b0c835529f686fcc2c8f030d4

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @OmarShehata!
* :heavy_check_mark: Signed CLA found.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by OmarShehata** [comment]
It appears as though the `gulpfile.js` in `cesium.com` branch is just behind. If you look at the diff between 1.67 and `cesium.com` you'll see `cesium.com` still says "AGI" instead of CesiumGS: 

https://github.com/CesiumGS/cesium/compare/1.67...cesium.com?diff=split

There are changes here that are intentional, I believe, like hard-coding the bucket names. But the biggest code difference is the way the `deployCesium` function has changed. The newer version starts with this `listAll` function which looks like it compares the files already uploaded before uploading any files.

**Comment by OmarShehata** [comment]
What I'm doing now is diffing the latest gulpfile.js from cesium.com: https://github.com/CesiumGS/cesium/blob/cesium.com/gulpfile.js

With the latest gulpfile.cjs from master: https://github.com/CesiumGS/cesium/blob/master/gulpfile.cjs

And selecting just the changes from cesium.com's gulpfile.js that relate to deploying on the `cesium.com` bucket.

**Comment by hpinkos** [comment]
@OmarShehata ping me once you have CI passing and I'll look at this

**Comment by hpinkos** [comment]
You can look here to see what's different between the `cesium.com` branch and master https://github.com/CesiumGS/cesium/compare/cesium.com

**Comment by OmarShehata** [comment]
Ok @hpinkos this should be ready now. Changes to `gulpfile.cjs` are now pretty minimal and include only what's necessary for the `js -> cjs` change. 

I tried to make it use the new `listAll` function which I believe fixes the issue of deleted files not getting deleted from S3, but it's a little more complicated because this branch needs to upload certain directories to different upload paths, whereas `master` just assumes everything goes in the same upload path. 

So as not to hold up the release, I think we should write an issue to update `master` so it allows you to specify a set of files to upload to which directory, and then this branch can just use that to avoid conflicts in the future.

**Comment by hpinkos** [comment]
> I tried to make it use the new listAll function which I believe fixes the issue of deleted files not getting deleted from S3

That's not an issue in this repo

**Comment by OmarShehata** [comment]
> That's not an issue in this repo

I must have missed where files on S3 but not in this repo get deleted.

If the upload code here works fine we could just get master to use it then. Either way it'll prevent issues like this in the future. I can write up that issue if that sounds good with you.

File: gulpfile.cjs [hunk 1]
```diff
@@ -583,6 +588,26 @@ function getMimeType(filename) {
     return { type: 'application/octet-stream', compress: true };
 }
 
+// get all files currently in bucket asynchronously
+function listAll(s3, bucketName, prefix, files, marker) {
```
Comment (hpinkos) [inline gulpfile.cjs:592]: remove this function since it's not used anywhere

Comment (OmarShehata) [reply gulpfile.cjs]: Done

**Pull request merged**